### DISCUSSION TITLE
Preserve state when opening nvim-tree with current_window=true.

### DISCUSSION THREAD

**Discussion by MKrbm** [nvim-tree/nvim-tree.lua#2419] [Q&A]
# How to reproduce

1. Split window with e.g. `vs`
2. Open nvim-tree by ` equire("nvim-tree.api").tree.toggle({current_window=true}) ` on one of the split window.
3. Close the nvim-tree by `q`
4. Then the window used for nvim-tree will be closed.

Expected behavious 
I want to just close the nvim-tree and go back to previous state of split windows. How can I configure this?

**Comment by alex-courtis** [comment]
`q` will close the window, following the usual behaviour of `:q` in a normal buffer in a window. Working as intended.

Suggestion: `:b#` instead of `q` to open the previous buffer in that window.

Alternatively, you might like to try: https://github.com/vim-scripts/bufkill.vim which allows:
* close the buffer without closing the window
* window specific MRU list - return to the previous buffer with `:BB` - more useful than `:b#`

I would be absolutely lost without the second.

**Reply by MKrbm** [reply]
Thank you very much! I will give it a try this weekend!
